cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go

```diff
@@ -0,0 +1,226 @@
+		Expect(err).ToNot(HaveOccurred())
+
+		numa0Device, err := findDeviceOnNUMANode(testingNode, sriovDevices, "0")
+		Expect(err).ToNot(HaveOccurred())
+		By("Using NUMA0 device " + numa0Device.Name)
+
+		numa1Device, err := findDeviceOnNUMANode(testingNode, sriovDevices, "1")
+		Expect(err).ToNot(HaveOccurred())
+		By("Using NUMA1 device " + numa1Device.Name)
+
+		// SriovNetworkNodePolicy
+		// NUMA node0 device excludeTopology = true
+		// NUMA node0 device excludeTopology = false
+		// NUMA node1 device excludeTopology = true
+		// NUMA node1 device excludeTopology = false
+
+		By("Create SRIOV policies and networks")
+
+		createSriovNetworkAndPolicy(
+			withNodeSelector(testingNode),
+			withNumVFs(8), withPfNameSelector(numa0Device.Name+"#0-3"),
+			withNetworkNameAndNamespace(sriovnamespaces.Test, "test-numa-0-exclude-topology-false"),
+			withExcludeTopology(false),
+		)
+
+		createSriovNetworkAndPolicy(
+			withNodeSelector(testingNode),
+			withNumVFs(8), withPfNameSelector(numa0Device.Name+"#4-7"),
+			withNetworkNameAndNamespace(sriovnamespaces.Test, "test-numa-0-exclude-topology-true"),
+			withExcludeTopology(true),
+		)
+
+		createSriovNetworkAndPolicy(
+			withNodeSelector(testingNode),
+			withNumVFs(8), withPfNameSelector(numa1Device.Name+"#0-3"),
+			withNetworkNameAndNamespace(sriovnamespaces.Test, "test-numa-1-exclude-topology-false"),
+			withExcludeTopology(false),
+		)
+
+		createSriovNetworkAndPolicy(
+			withNodeSelector(testingNode),
+			withNumVFs(8), withPfNameSelector(numa1Device.Name+"#4-7"),
+			withNetworkNameAndNamespace(sriovnamespaces.Test, "test-numa-1-exclude-topology-true"),
+			withExcludeTopology(true),
+		)
+
+		By("Waiting for SRIOV devices to get configured")
+		networks.WaitStable(sriovclient)
+	})
+
+	AfterAll(func() {
+		By("Cleaning performance profiles")
+
+		err := performanceprofile.CleanPerformanceProfiles()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Validate the creation of a pod with excludeTopology set to False and an SRIOV interface in a different NUMA node than the pod", func() {
+
+		pod := pods.DefinePod(sriovnamespaces.Test)
+		pods.RedefineWithGuaranteedQoS(pod, "1", "100m")
+		pod = pods.RedefinePodWithNetwork(pod, "test-numa-0-exclude-topology-false, test-numa-1-exclude-topology-false")
+
+		pod, err := client.Client.Pods(sriovnamespaces.Test).
+			Create(context.Background(), pod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			actualPod, err := client.Client.Pods(sriovnamespaces.Test).Get(context.Background(), pod.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(actualPod.Status.Phase).To(Equal(corev1.PodFailed))
+			g.Expect(actualPod.Status.Message).To(ContainSubstring("Resources cannot be allocated with Topology locality"))
+		}).Should(Succeed())
+	})
+})
+
+func findDeviceOnNUMANode(node *corev1.Node, devices []*sriovv1.InterfaceExt, numaNode string) (*sriovv1.InterfaceExt, error) {
```

Comment:
These (and below) are worth being in some pkg

Reply:
The below functions rely on test-specific default values. Moving them to a generic package would make them much less readable. WDYT?

Reply:
The only one I see test specific is createSriovNetworkAndPolicy.
Also, we already have functions to create sriov networks and sriov policies. I'd try to extend / modify those instead of having another variant scattered across the repo.